src/Grpc.Net.Client/GrpcChannel.cs

```diff
@@ -292,6 +292,15 @@ private static HttpHandlerContext CalculateHandlerContext(ILogger logger, Uri ad
                 }
             }
 
+            // Load balancing has been disabled on the SocketsHttpHandler.
+            if (socketsHttpHandler.Properties.TryGetValue("__GrpcLoadBalancingDisabled", out var value)
+                && value is bool loadBalancingDisabled && loadBalancingDisabled)
+            {
+                type = HttpHandlerType.Custom;
+                connectTimeout = null;
+                connectionIdleTimeout = null;
```

Comment:
What does this mean? That they won't be overridden on the handler?

Reply:
They have no meaning if the handler type isn't `SocketsHttpHandler`.

I've simplified the method to avoid doing this.